server/middlewares/checkReference.js

```diff
@@ -0,0 +1,20 @@
+/**
+  * This middleware intercepts the request and checks that the request
+  * contains a payment reference in the query parameter.
+  * @param {object} req the request object
+  * @param {object} res the response object
+  * @param {object} next the next middleware function
+
+  */
+const checkReference = (req, res, next) => {
```

Comment:
Is there a reason why this is implemented as a middleware instead of as a method in the controller? Given that it is used in just a single place I feel that it is better as a method in the PaymentController. This way everything about the payment is contained in a single place. (Single Responsibility Principle).

Reply:
Okay, I did that to avoid longer codes to in the validate controller, more or less like every aspect having its own purpose within the whole process.